### PR TITLE
Bug 1182762 — Add SimplePush statsd counters.

### PR DESCRIPTION
**Pull request by Natim**
https://bugzilla.mozilla.org/show_bug.cgi?id=1182762

 - [x] Add tests

### CODE REVIEW

**Comment by Natim** [comment]
@tarekziade if you want to do an early review while I am writing tests for that.

File: loop/routes/home.js [hunk 1]
```diff
@@ -45,6 +45,16 @@ module.exports = function(app, conf, logError, storage, tokBox) {
             request.put(req.query.SP_LOCATION, function(error, response) {
               pushStatus = (!error && response.statusCode === 200);
               returnStatus(storageStatus, requestError, pushStatus);
+              if (statsdClient !== undefined) {
+                statsdClient.count('loop.simplepush.call', 1);
+                var counter_push_status_counter;
+                if (pushStatus) {
+                  counter_push_status_counter = 'loop.simplepush.call.heartbeat.success';
+                } else {
+                  counter_push_status_counter = 'loop.simplepush.call.heartbeat.failures';
```
Comment (tarekziade) [inline loop/routes/home.js]: can we log the error/status code on errors ? that's useful for ops.  maye in the stderr?

File: loop/simplepush.js [hunk 2]
```diff
@@ -32,8 +32,16 @@ SimplePush.prototype = {
       request.put({
         url: simplePushUrl,
         form: { version: version }
-      }, function() {
-        // Catch errors.
+      }, function(err) {
+        if (self.statsdClient !== undefined) {
+          if (err) {
+            self.statsdClient.count("loop.simplepush.call.failures", 1);
```
Comment (tarekziade) [inline loop/simplepush.js]: same remark, lets log the error

File: loop/routes/home.js (hunk 1 above)
Comment (Natim) [inline loop/routes/home.js]: The problem is that we don't really care about this error and we don't want to have it logged in sentry. People can put whatever URLs they want here.

**Comment by Natim** [comment]
Updated.

**Comment by Natim** [comment]
Should we count for hostname as well?

File: loop/routes/home.js (hunk 1 above)
Comment (tarekziade) [inline loop/routes/home.js]: In production we do care if a Simple Push HTTP request fails with an error. It helps ops detecting what's wrong.

Comment (Natim) [inline loop/routes/home.js]: I have added the log here: https://github.com/mozilla-services/loop-server/pull/342/files#diff-3c8b9f512b8149de7980ce5ce9930be2R46

Do you want to log a Sentry error?

Comment (Natim) [inline loop/routes/home.js]: We may got a lot of spam if we do so.

**Comment by tarekziade** [comment]
> Do you want to log a Sentry error?
That sounds like a good idea

> We may got a lot of spam if we do so.

Why is that ?

1/ if we set a SP url it's supposed to work. If it does not it's a problem, no?
2/ Sentry aggregates the same error under a counter and sends a single event

**Comment by Natim** [comment]
> if we set a SP url it's supposed to work. If it does not it's a problem, no?

It is not the responsibility of the loop-server that the user configured a working SimplePush URL.

**Comment by tarekziade** [comment]
thx. r+

**Pull request merged**